src/main/scala/graphql/HanamuraServer.scala

```diff
@@ -0,0 +1,42 @@
+package graphql
+import caliban.schema.GenericSchema
+import caliban.GraphQL._
+import caliban.RootResolver
+import com.typesafe.config.ConfigFactory
+import mongodb.Mongo
+import org.http4s.implicits._
+import org.http4s.server.Router
+import org.http4s.server.blaze.BlazeServerBuilder
+import org.http4s.server.middleware.CORS
+import zio.{ RIO, ZIO }
+import zio.clock.Clock
+import zio.console.{ Console, putStrLn }
+import zio.interop.catz._
+object HanamuraServer extends CatsApp with GenericSchema[Console with Clock] {
+  val config = ConfigFactory.load()
+  val host   = config.getString("http.host")
+  val port   = config.getInt("http.port")
+  type HanamuraTask[A] = RIO[Console with Clock, A]
+  case class DatabaseConnection(conn: String = "test connection")
+  override def run(args: List[String]): ZIO[zio.ZEnv, Nothing, Int] =
+    (for {
+      service <- HanamuraService.make(Mongo.usersCollection)
+      interpreter = graphQL(
+        RootResolver(
+          Queries(() => service.sayHello),
+          Mutations(args => service.addUser(args))
```

Comment:
Your `User` class uses `ObjectId` which is not a standard type so caliban doesn't know how to handle it.

To solve this, you need to provide a `Schema` for it. You can easily re-use the schema for string:
```scala
implicit val objectIdSchema     = Schema.stringSchema.contramap[ObjectId](_.toString)
```

Also in mutations you pass User as an argument, so you need to provide an `ArgBuilder` for `ObjectId` to tell caliban how to parse such value:
```scala
implicit val objectIdArgBuilder = ArgBuilder.string.map(new ObjectId(_))
```

With these 2 additions, your PR compiles.
Related docs:
- schema: https://ghostdogpr.github.io/caliban/docs/schema.html#custom-types
- arg builder: https://ghostdogpr.github.io/caliban/docs/schema.html#arguments